core/embedding/colpali_embedding_model.py

```diff
@@ -15,21 +16,34 @@
 
 
 class ColpaliEmbeddingModel(BaseEmbeddingModel):
-    def __init__(self):
+    def __init__(self, model_name: str = "vidore/colqwen2.5-v0.2"):
         device = (
             "mps"
             if torch.backends.mps.is_available()
             else "cuda" if torch.cuda.is_available() else "cpu"
         )
-        self.model = ColQwen2.from_pretrained(
-            "vidore/colqwen2-v1.0",
-            torch_dtype=torch.bfloat16,
-            device_map=device,  # Automatically detect and use available device
-            attn_implementation="flash_attention_2" if device == "cuda" else "eager",
-        ).eval()
-        self.processor: ColQwen2Processor = ColQwen2Processor.from_pretrained(
-            "vidore/colqwen2-v1.0"
-        )
+        if model_name == "vidore/colqwen2.5-v0.2":
```

Comment:
[nit] we generally prefer having match-case style statements for model/config selection. Totally optional, tho.